tsdb/block_test.go

```diff
@@ -418,6 +418,84 @@ func BenchmarkLabelValuesWithMatchers(b *testing.B) {
 	}
 }
 
+func TestLabelNamesWithMatchers(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "test_block_label_names_with_matchers")
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, os.RemoveAll(tmpdir))
+	}()
```

Comment:
We can use the `t.Cleanup` instead of defer. Also for the defers below.

Reply:
I followed the same approach as in the rest of the tests. May I change that as a separate PR to avoid scope creep?

Reply:
Done: https://github.com/prometheus/prometheus/pull/9083/commits/8bf71bd875bc4f3cfdf1497bf10be56e73468343